### PR TITLE
adding basic and none authentication types to the apis

### PR DESCRIPTION
**Pull request by obadrawi**
adding basic and none authentication types as an element in the varkes_config files for every api (Issue #153)

### CODE REVIEW

File: app-connector-client/server/config.js [hunk 1]
```diff
@@ -56,10 +57,26 @@ function configValidation(configJson) {
             }
         }
     }
+    if (apis) {
+        for (var i = 1; i <= apis.length; i++) {
+            {
+                var api = apis[i - 1];
+
+                if (!api.authtype) {
+                    error_message += "\napi " + (api.name ? api.name : "number " + i) + ": missing attribute 'authtype', authtype is mandatory";
```
Comment (a-thaler) [inline app-connector-client/server/config.js]: We should have a default instead, maybe "none"?

File: app-connector-client/server/config.js [hunk 2]
```diff
@@ -56,10 +57,26 @@ function configValidation(configJson) {
             }
         }
     }
+    if (apis) {
+        for (var i = 1; i <= apis.length; i++) {
+            {
+                var api = apis[i - 1];
+
+                if (!api.authtype) {
```
Comment (a-thaler) [inline app-connector-client/server/config.js]: "authtype" does not look nice... what about simple "auth"?

File: app-connector-client/test/varkes_config.json [hunk 3]
```diff
@@ -41,9 +42,9 @@
         {
             "baseurl": "/entity/v1",
             "metadata": "/metadata",
+            "authtype": "basic",
```
Comment (a-thaler) [inline app-connector-client/test/varkes_config.json]: Please write a test case!

File: app-connector-client/server/config.js (hunk 1 above)
Comment (obadrawi) [reply app-connector-client/server/config.js]: no check for auth element exists.. if doesn't exist the default is none

File: app-connector-client/server/config.js (hunk 2 above)
Comment (obadrawi) [reply app-connector-client/server/config.js]: changed to auth

File: app-connector-client/server/config.js [hunk 4]
```diff
@@ -56,10 +57,23 @@ function configValidation(configJson) {
             }
         }
     }
+    if (apis) {
+        for (var i = 1; i <= apis.length; i++) {
+            {
+                var api = apis[i - 1];
+
+                if (api.auth && !(api.auth == "oauth" || api.auth == "none" || api.auth == "basic")) {
+                    error_message += "\napi " + (api.name ? api.name : "number " + i) + ": attribute 'authtype' should be one of three values [oauth, basic, none]";
```
Comment (a-thaler) [inline app-connector-client/server/config.js]: please replace authtype in message

File: app-connector-client/server/config.js [hunk 5]
```diff
@@ -56,10 +57,23 @@ function configValidation(configJson) {
             }
         }
     }
+    if (apis) {
+        for (var i = 1; i <= apis.length; i++) {
+            {
```
Comment (a-thaler) [inline app-connector-client/server/config.js]: please remove on bracket

File: app-connector-client/server/config.js [hunk 6]
```diff
@@ -56,10 +57,23 @@ function configValidation(configJson) {
...
+                }
+
+            }
+        }
+    }
     if (configJson.logo && !configJson.logo.match(/^.+\.(svg)$/)) {
         error_message += "\nlogo image must be in svg format"
     }
     if (error_message != "") {
+
```
Comment (a-thaler) [inline app-connector-client/server/config.js]: please avoid unneeded new lines

File: examples/combined-odata-mock/varkes_config.json [hunk 7]
```diff
@@ -4,6 +4,7 @@
         {
             "name": "services",
             "specification": "apis/services.xml",
+            "authtype": "basic",
```
Comment (a-thaler) [inline examples/combined-odata-mock/varkes_config.json]: adjust authtype

File: examples/combined-openapi-mock/varkes_config.json [hunk 8]
```diff
@@ -4,14 +4,16 @@
         {
             "baseurl": "/api2",
             "specification": "apis/schools.yaml",
-            "name": "Schools API"
+            "name": "Schools API",
+            "authtype": "basic"
```
Comment (a-thaler) [inline examples/combined-openapi-mock/varkes_config.json]: adjust authtype

Comment (obadrawi) [reply examples/combined-openapi-mock/varkes_config.json]: Done

File: examples/combined-odata-mock/varkes_config.json (hunk 7 above)
Comment (obadrawi) [reply examples/combined-odata-mock/varkes_config.json]: Done

File: app-connector-client/server/config.js (hunk 6 above)
Comment (obadrawi) [reply app-connector-client/server/config.js]: done

File: app-connector-client/server/config.js (hunk 5 above)
Comment (obadrawi) [reply app-connector-client/server/config.js]: done

File: app-connector-client/server/config.js (hunk 4 above)
Comment (obadrawi) [reply app-connector-client/server/config.js]: Done

**Pull request merged**